crates/holochain/src/core/state/cascade.rs

```diff
@@ -678,22 +763,55 @@ where
                         .expect("Status is live but no headers?");
 
                     // We have an oldest live header now get the element
-                    CascadeResult::Ok(
+                    CascadeResult::Ok(Some(
                         self.get_element_local_raw(&oldest_live_header.header_hash)?
                             .map(Search::Found)
                             // It's not local so check the network
                             .unwrap_or(Search::Continue(oldest_live_header.header_hash)),
-                    )
+                    ))
                 }
                 EntryDhtStatus::Dead
                 | EntryDhtStatus::Pending
                 | EntryDhtStatus::Rejected
                 | EntryDhtStatus::Abandoned
                 | EntryDhtStatus::Conflict
                 | EntryDhtStatus::Withdrawn
-                | EntryDhtStatus::Purged => CascadeResult::Ok(Search::NotInCascade),
+                | EntryDhtStatus::Purged => CascadeResult::Ok(None),
             }
-        })?;
+        })
+    }
+
+    #[instrument(skip(self, options))]
+    /// Returns the oldest live [Element] for this [EntryHash] by getting the
+    /// latest available metadata from authorities combined with this agents authored data.
+    pub async fn dht_get_entry(
+        &mut self,
+        entry_hash: EntryHash,
+        options: GetOptions,
+    ) -> CascadeResult<Option<Element>> {
+        debug!("in get entry");
+        // Update the cache from the network
+        self.fetch_element_via_entry(entry_hash.clone(), options.clone())
+            .await?;
+
+        let cache_data = ok_or_return!(self.cache_data.as_ref(), None);
+        let authored_data = ok_or_return!(self.authored_data.as_ref(), None);
+        // Meta Cache
+        let oldest_live_element =
+            match self.get_oldest_live_element(&entry_hash, &DbPair::from(cache_data))? {
+                // Look for the element data in authored
+                Some(Search::Continue(oldest_live_header)) => {
+                    match authored_data.element.get_element(&oldest_live_header)? {
+                        Some(element) => Search::Found(element),
```

Comment:
I'm not sure about this logic. If you search the cache first, and it is mostly clear, but has a much newer live element, won't it return that, even though the authored_data might have an older live element?

Reply:
hmm good catch, seems I need to actually gather all the elements together to search from both stores